File: apps/institution-api/Models/Dto/SpecializationDto.cs
using System;

namespace ProviderApi.Models.Dto
{
  public class SpecializationDto
  {
    public int SpecializationId { get; set; }
    public int? SpecializationNumber { get; set; }
    public string SpecializationName { get; set; }
    public string SpecializationCode { get; set; }
    public string SpecializationStatus { get; set; }
    public int CipCodeId { get; set; }
    public string CipSubSeriesCode { get; set; }
    public string CipSubSeries { get; set; }
    public string AdministrativeUnitCode { get; set; }
    public string AdministrativeUnit { get; set; }
    public string EntranceRequirementCode { get; set; }
    public string EntranceRequirement { get; set; }
    public string UnitOfLoad { get; set; }
    public string PrimaryFundingCode { get; set; }
    public int? ProgramId { get; set; }
    public DateTime? BusinessStartDate { get; set; }
    public DateTime? BusinessEndDate { get; set; }
  }
}

File: libs/common/models/specialization-dto.ts



export class SpecializationDto {
    public specializationId: number = 0;
    public specializationNumber: number;
    public specializationName: string = '';
    public specializationCode: string = '';
    public specializationStatus: string = '';
    public cipCodeId: number = 0;
    public cipSubSeriesCode: string = '';
    public cipSubSeries: string = '';
    public administrativeUnitCode: string = '';
    public administrativeUnit: string = '';
    public entranceRequirementCode: string = '';
    public entranceRequirement: string = '';
    public unitOfLoad: string = '';
    public primaryFundingCode: string = '';
    public programId: number;
    public businessStartDate: Date;
    public businessEndDate: Date;

    constructor(init? : Partial<SpecializationDto>) {
        
        Object.assign(this, init);
    }
}

